db/utils.go

```diff
@@ -0,0 +1,15 @@
+package db
+
+import (
+	"time"
+
+	"gorm.io/gorm"
+)
+
+// Model is a copy of the gorm model type but using slightly different configurations
+type Model struct {
+	ID        uint      `gorm:"primarykey"`
+	CreatedAt time.Time `gorm:"index"` // this is changed to include the index
```

Comment:
i might need to undo this index, as it seems to have made queries slower which is strange, i thought indexes would speed up queries but i guess not.